src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs

```diff
@@ -173,6 +173,9 @@ public override void ExecuteCmdlet()
         {
             ExecutionBlock(() =>
             {
+                //check if DefaultContext existed, PSInvalidOperationException will be thrown
+                var CheckDefaultContext = DefaultContext;
```

Comment:
this is never used

Reply:
this is a safe guard, if user was not login, this line will protect from null reference